src/extension.ts

```diff
@@ -105,11 +106,36 @@ export async function deactivate() {
   const now = performance.now();
   const data = {
     name: "sessionStatus",
-    time: Math.round(now - activatedTimestamp)
+    time: Math.round(now - activatingTimestamp)
   }
   if (cleanJavaWorkspaceIndicator && fs.existsSync(cleanJavaWorkspaceIndicator)) {
     data.name = "cleanJavaLSWorkspace";
   }
   sendInfo("", data);
   await disposeTelemetryWrapper();
 }
+
+async function recordStartupTime() {
+  const javaExt = vscode.extensions.getExtension("redhat.java");
+  // only count the e2e time when java extension is not activated at the moment.
+  // if it's already activated, we are not clear if it is activated with pack at
+  // the same moment.
+  if (javaExt && !javaExt.isActive) {
+    const delay = promisify(setTimeout);
+    const timeout = 1 * 60 * 1000; // wait 1 min at most
+    let count = 0;
+    while(!javaExt.isActive && count < timeout) {
+      await delay(1000);
+      count += 1000;
+    }
```

Comment:
The condition branch `if (javaExt && !javaExt.isActive)` should end at L130.

Reply:
I was intended to do this, while not sure if this makes enough sense. What do you think about the follow case:

1. In an empty workspace, user triggers some `redhat.java` specific command. such as
![image](https://github.com/microsoft/vscode-java-pack/assets/6193897/e98dbe2f-16f2-4a7e-b33d-fdc4639b8e40) to open a webview.
2. Then, `redhat.java` will be activated, while pack is not.

So the purpose here is to make sure that two extensions are almost activated at the same time.